packages/fiori/src/MediaGallery.ts

```diff
@@ -260,12 +78,189 @@ const metadata = {
+	 *
+	 * @type {boolean}
+	 * @name sap.ui.webc.main.MediaGallery.prototype.showAllThumbnails
+	 * @defaultvalue false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	showAllThumbnails!: boolean;
+
+	/**
+	 * If enabled, a <code>display-area-click</code> event is fired
+	 * when the user clicks or taps on the display area.
+	 * <br>
+	 * The display area is the central area that contains
+	 * the enlarged content of the currently selected item.
+	 *
+	 * @type {boolean}
+	 * @name sap.ui.webc.main.MediaGallery.prototype.interactiveDisplayArea
+	 * @defaultvalue false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	interactiveDisplayArea!: boolean;
+
+	/**
+	 * Determines the layout of the component.
+	 * <br><br>
+	 * Available options are:
+	 * <ul>
+	 * <li><code>Auto</code></li>
+	 * <li><code>Vertical</code></li>
+	 * <li><code>Horizontal</code></li>
+	 * </ul>
+	 *
+	 * @type {sap.ui.webc.fiori.types.MediaGalleryLayout}
+	 * @name sap.ui.webc.main.MediaGallery.prototype.layout
+	 * @defaultvalue "Auto"
+	 * @public
+	 */
+	@property({ type: MediaGalleryLayout, defaultValue: "Auto" })
+	layout!: MediaGalleryLayout;
+
+	/**
+	 * Determines the horizontal alignment of the thumbnails menu
+	 * vs. the central display area.
+	 * <br><br>
+	 * Available options are:
+	 * <ul>
+	 * <li><code>Left</code></li>
+	 * <li><code>Right</code></li>
+	 * </ul>
+	 *
+	 * @type {sap.ui.webc.fiori.types.MediaGalleryMenuHorizontalAlign}
+	 * @name sap.ui.webc.main.MediaGallery.prototype.menuHorizontalAlign
+	 * @defaultvalue "Left"
+	 * @public
+	 */
+	@property({ type: MediaGalleryMenuHorizontalAlign, defaultValue: MediaGalleryMenuHorizontalAlign.Left })
+	menuHorizontalAlign!: MediaGalleryMenuHorizontalAlign;
+
+	/**
+	 * Determines the vertical alignment of the thumbnails menu
+	 * vs. the central display area.
+	 * <br><br>
+	 * Available options are:
+	 * <ul>
+	 * <li><code>Top</code></li>
+	 * <li><code>Bottom</code></li>
+	 * </ul>
+	 *
+	 * @type {sap.ui.webc.fiori.types.MediaGalleryMenuVerticalAlign}
+	 * @name sap.ui.webc.main.MediaGallery.prototype.menuVerticalAlign
+	 * @defaultvalue "Bottom"
+	 * @public
+	 */
+	@property({ type: MediaGalleryMenuVerticalAlign, defaultValue: MediaGalleryMenuVerticalAlign.Bottom })
+	menuVerticalAlign!: MediaGalleryMenuVerticalAlign;
+
+	/**
+	 * Determines the actual applied layout type
+	 * (esp. needed when the app did not specify a fixed layout type
+	 * but selected <code>Auto</code> layout type).
+	 * <br><br>
+	 * Possible values are:
+	 * <ul>
+	 * <li><code>Vertical</code></li>
+	 * <li><code>Horizontal</code></li>
+	 * </ul>
+	 *
+	 * @type {sap.ui.webc.fiori.types.MediaGalleryLayout}
+	 * @defaultvalue "Vertical"
+	 * @private
+	 */
+	@property({ type: MediaGalleryLayout, defaultValue: "Vertical" })
+	effectiveLayout!: MediaGalleryLayout;
+
+	/**
+	 * Defines the current media query size.
+	 *
+	 * @private
+	 */
+	@property()
+	mediaRange!: string;
```

Comment:
This could be described more precisely with a type as we know the exact possible values - could not be any string

Reply:
I discussed this with Nayden and we agreed that should be done as a separate task about control refactoring.

Reply:
Yes, we've discussed it to be changed in another iteration because mediaRange is implemented in the following way everywhere the MediaRange util is used.

Reply:
Ok